compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -849,15 +845,15 @@ class TreeUnpickler(reader: TastyReader,
     }
 
     def skipToplevel()(implicit ctx: Context): Unit= {
-      if (!isAtEnd)
-        nextByte match {
-          case IMPORT | PACKAGE =>
-            skipTree()
-            skipToplevel()
-          case _ =>
-        }
+      if (!isAtEnd && isTopLevel) {
+        skipTree()
+        skipToplevel()
+      }
     }
 
+    def isTopLevel(implicit ctx: Context): Boolean =
+      nextByte == IMPORT || nextByte == PACKAGE
+
     def readTopLevel()(implicit ctx: Context): List[Tree] = {
       @tailrec def read(acc: ListBuffer[Tree]): List[Tree] = nextByte match {
         case IMPORT | PACKAGE =>
```

Comment:
Could use isTopLevel here. Also I wonder if we couldn't simplify that to return a single Tree instead of a List[Tree] if we never pickle a list of trees.

Reply:
We may have problems tasty files that have top-level imports

Reply:
If I put a top level import I get

```scala
import scala.collection.mutable.TreeMap
...
```

```scala
     0: PACKAGE(80)
     2:   TERMREFpkg 1 [<empty>]
     4:   IMPORT(8)
     6:     SELECT 2 [mutable]
     8:       SELECT 3 [collection]
    10:         TERMREFpkg 4 [scala]
    12:     IMPORTED 5 [TreeMap]
    14:   PACKAGE(66)
    16:     TERMREFpkg 6 [foo]
    ...
```

same with 
```scala
package bar
import scala.collection.mutable.TreeMap
```

```scala
     0: PACKAGE(80)
     2:   TERMREFpkg 1 [bar]
     4:   IMPORT(8)
     6:     SELECT 2 [mutable]
     8:       SELECT 3 [collection]
    10:         TERMREFpkg 4 [scala]
    12:     IMPORTED 5 [TreeMap]
```

Reply:
I think we may go further and do 

```scala
def isTopLevel(implicit ctx: Context): Boolean = 
  nextByte == PACKAGE
```